### PR TITLE
Add duplicate checking for public address string in AddPA command

### LINKED ISSUE #235: [PE-D][Tester B] Duplicate public addess and wallet_name/label allowed for different network
Similar the the duplicate public address allowed for differnt wallet_name/label one, I was able to add the same wallet and public address under a different network

If this was the intended feature and allowed, do ignore this bug report!

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: The0nlyJuan/ped#9

### PR DESCRIPTION
**Pull request by chongtzezhao**
fixes #235 #226 #227 #222 #211 #247

### CODE REVIEW

**Review by Airiinnn — APPROVED** [review]
LGTM, necessary validity check as current behavior causes software to misbehave

File: src/main/java/seedu/address/logic/commands/AddPublicAddressCommand.java [hunk 1]
```diff
@@ -77,6 +78,25 @@ private static Person mergePersons(Person personToEdit, EditPersonDescriptor edi
...
     public CommandResult execute(Model model) throws CommandException {
+
+        // check if the public address already exists
+
+        String paToAdd = editPersonDescriptor.getPublicAddresses()
+                .map(PublicAddressesComposition::getOnePublicAddress)
+                .orElseThrow(() -> new CommandException("Public Address to add is missing"))
+                .getPublicAddressString();
+
+        List<String> allPAs = model.getFilteredPersonList().stream()
```
Comment (Airiinnn) [inline src/main/java/seedu/address/logic/commands/AddPublicAddressCommand.java:89]: Get all persons instead of just filter list

**Comment by chongtzezhao** [comment]
image

**Review by Nicholascyx — APPROVED** [review]
LGTM

**Review by travisim — APPROVED** [review]
LGTM

**Pull request merged**